ray-operator/controllers/ray/raycluster_controller_test.go

```diff
@@ -463,6 +498,46 @@ var _ = Context("Inside the default namespace", func() {
 			})
 			Expect(err).NotTo(HaveOccurred(), "Failed to update RayCluster")
 
+			// Make sure the suspending process is atomic by removing finalizers one by one.
+			if withConditionEnabled {
+				// Make sure the cluster is now marked with RayClusterSuspending.
+				Eventually(findRayClusterSuspendStatus, time.Second*3, time.Millisecond*500).Should(Equal(rayv1.RayClusterSuspending))
+				// Then turn the Spec.Suspend to false.
+				err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Can you create a function for updating `spec.Suspend` in case we call it twice? Maybe you can refer to `updateRayJobSuspendField`.